fb/xrgb.go

```diff
@@ -0,0 +1,109 @@
+// Copyright ©2016 The ev3go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package fb
+
+import (
+	"errors"
+	"image"
+	"image/color"
+	"image/draw"
+)
+
+// NewXRGB returns a new XRGB image with the given bounds.
+func NewXRGB(r image.Rectangle) *XRGB {
+	w, h := r.Dx(), r.Dy()
+	stride := 2 * w
+	pix := make([]uint8, stride*h)
+	return &XRGB{Pix: pix, Stride: stride, Rect: r}
+}
+
+// NewXRGBWith returns a new XRGB image with the given bounds,
+// backed by the []byte, pix. If stride is zero, a working stride
+// is computed. If the length of pix is less than stride*h, an
+// error is returned.
+func NewXRGBWith(pix []byte, r image.Rectangle, stride int) (draw.Image, error) {
+	w, h := r.Dx(), r.Dy()
+	if stride == 0 {
+		stride = 4 * w
+	}
+	if len(pix) < stride*h {
+		return nil, errors.New("ev3dev: bad pixel buffer length")
+	}
+	return &XRGB{Pix: pix, Stride: stride, Rect: r}, nil
+}
+
+// XRGB is an in-memory image whose At method returns Pixel565 values.
+type XRGB struct {
+	// Pix holds the image's pixels, as 32 bit XRGB
+	// values stored in little-endian order.
+	// The pixel at (x, y) is the four bytes at
+	// Pix[2*(x-Rect.Min.X) + (y-Rect.Min.Y)*Stride].
+	Pix []uint8
+	// Stride is the Pix stride (in bytes) between
+	// vertically adjacent pixels.
+	Stride int
+	// Rect is the image's bounds.
+	Rect image.Rectangle
+}
+
+// ColorModel returns the XRGB color model.
+func (p *XRGB) ColorModel() color.Model { return RGB565Model }
+
+// Bounds returns the bounding rectangle for the image.
+func (p *XRGB) Bounds() image.Rectangle { return p.Rect }
+
+// At returns the color of the pixel at (x, y).
+func (p *XRGB) At(x, y int) color.Color {
+	if !(image.Point{x, y}.In(p.Rect)) {
+		return color.RGBA{}
+	}
+	i := p.pixOffset(x, y)
+	return PixelXRGB{R: p.Pix[i+2], G: p.Pix[i+1], B: p.Pix[i]}
+}
+
+// Set sets the color of the pixel565 at (x, y) to c.
```

Comment:
Again, the mention of pixel565 makes me think I am missing something here.

Reply:
No, this is just me. Sorry.